src/main/java/com/acme/banking/dbo/domain/SavingAccount.java

```diff
@@ -8,6 +8,10 @@
     private double amount;
 
     public SavingAccount(UUID id, Client client, double amount) {
+        if (id == null) { throw new IllegalArgumentException("id"); }
+        if (client == null) { throw new IllegalArgumentException("client"); }
+        if (amount >= 0) { throw new IllegalArgumentException("amount"); }
```

Comment:
Here we should use `amount < 0` clause ofc. It's just a typo. (or maybe we should start conversation about business rules about this).